### ISSUE TITLE
Confusion about XDG Base Directory specification and what is $XDG_DATA_DIR?

### ISSUE THREAD

**Issue by HimDek** [cantino/mcfly#269]
[Database Location](https://github.com/cantino/mcfly#database-location) in the README says `$XDG_DATA_DIR/mcfly/history.db` which defaults to `~/.local/share/mcfly/history.db` would be used as the location for McFly's SQLite database if `~/.mcfly/` doesnot exist.

However, there is no such environment variable named `$XDG_DATA_DIR` specified neither in the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) nor in the [Arch Wiki](https://wiki.archlinux.org/title/XDG_Base_Directory) on the same topic. The one that is the specification is `$XDG_DATA_DIRS` whose default value is `/usr/local/share/:/usr/share/`, which clearly is not the one applicable here.

In my system, following the XDG Base Directory Specification, I have `$XDG_DATA_HOME` set to `$HOME/.local/share/`, but  `$XDG_DATA_DIR` is not set. However, after I moved the SQLite database file of McFly to `~/.local/share/mcfly/history.db`, it seems to be working as expected.

**NOTE:** When I installed McFly, I probably did not have `$XDG_DATA_HOME` set but now I do.

So, which environment variable does McFly exactly use for its database location?

**Comment by cantino** [comment]
@Awernx do you have any insight into this?

**Comment by o6uoq** [comment]
@HimDek are you using **macOS**?

**Comment by HimDek** [comment]
> @HimDek are you using **macOS**?

No. I am using Arch Linux.

**Comment by nedbat** [comment]
I agree that XDG_DATA_DIR is not a correct environment variable. In addition, now the color settings are looked up there, but surely configuration should be in XDG_CONFIG_HOME?

**Comment by cantino** [comment]
Unfortunately I don't know about this, I trusted the original PR. If it's using an incorrect environment variable, we should fix it.

**Comment by nedbat** [comment]
Good to know that it's only the docs with the misspelled environment variable name, but I still think it's the wrong XDG directory.  Configuration should be in CONFIG, not DATA.

**Comment by torculus** [comment]
@cantino , there's [another project](https://github.com/Veirt/weathr/blob/a91a603b344eeab39718cb9ec21600c80cd5ca2f/src/config.rs#L89) that uses the `dirs` crate to specifically load the config dir in a platform-independent way. I'll try to explore a way to incorporate that here and open a pull request

**Comment by cantino** [comment]
Thank you @torculus, appreciate any help here!
